packages/shared_preferences/shared_preferences/README.md

```diff
@@ -85,6 +85,25 @@ If this is problematic for your use case, you can thumbs up
 interest in APIs that provide direct (asynchronous) access to the underlying
 preference store, and/or subscribe to it for updates.
 
+### Migration and Prefixes
+
+By default, the `SharedPreferences` plugin will only read (and write) preferences
+that begin with the prefix `flutter.`. This is all handled internally by the plugin
+and does not require manually adding this prefix.
+
+If preferences are being migrated from a non-Flutter version of an app,
+they will need to be transformed to include the prefix, or they will not be
+available in the preferences provided by `SharedPreferences`.
```

Comment:
I think this may confuse people, because it's not clear that this isn't something that can be done via the plugin (and would instead require custom native code). I would remove this, and focus on explaining how to approach this case using the plugin and an empty prefix.